index.js

```diff
@@ -30,6 +35,29 @@ app.get('/ideas', (req, res) => {
     res.status(200).json(getIdeas());
 });
 
+//DONATION METHODS
+
+app.get('/donations', (req, res) => {
+    res.status(200).json(ideasDonationSum(req.body.ideaId));
+});
+
+app.get('/donations/sumDifference', (req, res) => {
+    res.status(200).json(ideasSumDifference(req.body.ideaId));
+});
+
+app.post('/donations', (req, res) => {
+    res.status(200)
+        .json(
+            createDonationsWithDataValidation(
```

Comment:
Can `createDonationsWithDataValidation` function throw an error?

Reply:
Updated and tested.